test/gui/blueprint.lua

```diff
@@ -202,6 +193,43 @@ end
 
 --auto enter and leave cursor-supporting mode
 
+function test.restore_mode()
+    df.global.ui.main.mode = df.ui_sidebar_mode.Stockpiles
```

Comment:
I know switching to some modes directly with this approach can cause crashes (e.g. `q` mode skips DF's logic to initialize `selected_building`, which could be garbage). Could this be replaced with `send_keys('D_STOCKPILES')`?

Reply:
I hadn't done it that way since it presupposes that the mode is in `Default`, which could make the test flaky if whatever test runs before it doesn't clean up properly. This is a non-issue for the first test in a suite since the test harness ensures the mode is in `Default` for fortress-mode tests, but individual tests within a suite do not guarantee that we end up in the same mode we started with.

I also thought there wasn't a risk of crash here because of the modes I chose (stockpiles here and burrows below) were "non-dangerous" (I specifically stayed away from `q`) so I didn't bother to copy in (or call) the generic mode-switching code in `gui/blueprint`.

Your point is well taken, though, that forcing a mode switch has proven to be trouble and we shouldn't encourage it. I'll move the generic mode-switching code to `dwarfmode.lua` and modify these tests to call it.

Reply:
done